### PR TITLE
Add findFileAsync helper

### PR DESCRIPTION
**Pull request by steelbrain**
cc @kepler0 @noseglid @Arcanemagus @SpainTrain

[Review on Reviewable](https://reviewable.io/reviews/atomlinter/atom-linter/66)

### CODE REVIEW

File: lib/helpers.coffee [hunk 1]
```diff
@@ -158,6 +158,34 @@ module.exports = Helpers =
...
+      names = [names]
+
+    chunks = directory.split(path.sep)
+    promise = Promise.resolve(null)
+
+    while chunks.length
+      currentDir = chunks.join(path.sep)
+      break unless currentDir
+
+      do (currentDir) ->
```
Comment (steelbrain) [inline lib/helpers.coffee]: I dislike cofee-script so much! It could've been a simple `let` in ES6

Comment (kepler0) [inline lib/helpers.coffee]: Couldn't you write it in ES6 and require it in the main file?

Comment (steelbrain) [inline lib/helpers.coffee]: After this is done, I'll be rewriting this whole file into ES6. We got no reason to stick to coffee anymore, Babel rocks :metal:

Comment (iam4x) [inline lib/helpers.coffee]: @steelbrain hahaha, I wished you thought this when the re-wrote happened!

Comment (steelbrain) [inline lib/helpers.coffee]: @iam4x Babel has evolved a lot lately and It's fixed the things I disliked about it, especially with Babel 6.0 the modular transforms are a blast :bomb:

Comment (kepler0) [inline lib/helpers.coffee]: You could just write this function in another file and require it in the CoffeeScript file now, so you don't have to rewrite this new function later.

Comment (steelbrain) [inline lib/helpers.coffee]: I don't think that's the best way to do it. Now does anybody got anything against merging this?

Comment (iam4x) [inline lib/helpers.coffee]: nope it's good :+1:

**Comment by kepler0** [comment]
We really should've waited for @noseglid's okay before merging...

**Comment by steelbrain** [comment]
@kepler0 The tests are passing, the API is the same, what more could one want :wink:

**Comment by noseglid** [comment]
Looks good from what I can tell. Don't know much and don't care for CoffeeScript so can't give any specifics about that. Now use it in `linter-javac`! :)

**Pull request merged**